src/mats_utils/rawdata/read_data.py

```diff
@@ -113,10 +117,42 @@ def read_MATS_payload_data(start_date,end_date,data_type='HTR',filter=None,versi
     if end_date.tzinfo == None:
         end_date = end_date.replace(tzinfo=timezone.utc)
 
+    partitioning = ds.partitioning(
+        schema=pa.schema(
+            [
+                ("year", pa.int16()),
+                ("month", pa.int8()),
+                ("day", pa.int8()),
+            ]
+        ),
+    )
+
     dataset = ds.dataset(
         file,
         filesystem=s3,
+        partitioning=partitioning,
         )
+    
+
+    start_with_margin =  start_date - timedelta(hours=1)
+    stop_with_margin = end_date + timedelta(hours=1)
```

Comment:
I assume this is taken from the corresponding function for reading L1A/B where partitioning is by hour? Adding one day is probably better for this partitioning, though 1 hour might be OK for the start. It might be OK the way it is, but make sure you think about it. so that it has the intended effect.

Reply:
changed the padding to 1 day